### ISSUE TITLE
CSS default styling should not allocate mem for some common defaults

### ISSUE THREAD

**Issue by npillmayer** [npillmayer/tyse#8]
Some very common CSS defaults values (e.g., "color") are inherited and therefore should always resolve to a valid value (color.Color in this case). This leads to an instantiated value for every box. That's OK for boxes which do have other styles as well, as a style type has then to be allocated anyway. But we should not allocate anything for boxes which do only have common inherited style defaults (which is presumably a very frequent case).

My current implementation idea is to have the user-agent defaults for some styles ("color", "font-family") set to "" (empty string). This will suppress creating a style container.

The flip side of this implementation is that the styling engine has to stop and think in case of empty styles.
Perhaps it would be wise to instantiate them before rendering? But then we should have done it before layout, as we will need the same mem size anyway.

Basically it's a trade-off between code-complication+performance vs memory footprint.

Affected files:
* dom/style/defaults.go    (style defaults)
* frame/boxtree/generate.go  (generation of box-tree)
and other files during rendering

**Comment by npillmayer** [comment]
Defined a user-agent value for non-inherited properties (like "color", "border-top-color", …) of "default".
Property color converter in

* dom/style/convert.go

will treat this as a null value, wheras "" (NullProperty) will be treatet as a hint to return a UA value.
Works fine so far. Just will have to remember the semantics of "default".

**Comment by npillmayer** [comment]
Documented in dom/style/defaults.go
